dask_ml/decomposition/incremental_pca.py

```diff
@@ -127,14 +127,21 @@ def __init__(
         self,
         n_components=None,
         whiten=False,
+        center=True,
         copy=True,
         batch_size=None,
         svd_solver="auto",
         iterated_power=0,
         random_state=None,
     ):
+        if center is False:
```

Comment:
I *think* that the scikit-learn API wants all validation of parameters to be deferred until `.fit` time. So this check could go at the top of `.fit`.

Reply:
Yes, this makes sense. I've had a read over the scikit-learn codebase and it seems they've got a mixture of checks - some upon `__init__`, but _mostly_ upon `fit`.
For reference, some of the exceptions where checks are performed upon `__init__`:
- [decomposition/_kernel_pca.py](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/decomposition/_kernel_pca.py#L151) (commit from 2012-02-17, 9c6e0445ae)
- [decomposition/_fastica.py](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/decomposition/_fastica.py#L400) (commit from 2018-04-04, 1a3201cc30)
- [feature_extraction/text.py](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/feature_extraction/text.py#L1027) (commit from 2013-03-14, 3444bfbe88)
- [svm/_base.py](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/svm/_base.py#L81) (commit from 2020-02-16, 9b39c4c4d2)

Seems it's not necessarily been a recent guideline in their API, but I confirm I'm happy to update this PR accordingly.

P.S.: It seems the the scikit-learn codebase is not very consistent between `_check_params` and `_check_parameters` either. Sometimes either of these is utilized for validation, and in other instances - validation is performed directly as part of `_fit`/`fit` (I appreciate that that's saving one extra method call, but still...).

Reply:
`scikit-learn`'s documentation is very c learn about this: validation should not happen at initialization time.

From [the "initialization" section in "Developing scikit-learn estimators"](https://scikit-learn.org/stable/developers/develop.html#instantiation):

> There should be no logic, not even input validation, and the parameters should not be changed.

And from [the "parameters and init" section](https://scikit-learn.org/stable/developers/develop.html#parameters-and-init)

> The easiest and recommended way to accomplish this is to not do any parameter validation in `__init__`. All logic behind estimator parameters, like translating string arguments into functions, should be done in `fit`.

Reply:
Noted, thanks!